pkg/controller/syncmachineset/gcpactuator_test.go

```diff
@@ -51,11 +50,9 @@ func TestGCPAcuator(t *testing.T) {
 					instanceType: "testInstance",
 				},
 			}),
-			mockGCPClient: func(mockCtrl *gomock.Controller) *mockgcp.MockClient {
-				client := mockgcp.NewMockClient(mockCtrl)
-				mockListComputeImage(client, []string{"testImage"})
-				mockListComputeZones(client, []string{"zone1"})
-				return client
+			mockGCPClient: func(client *mockgcp.MockClient, cd *hivev1.ClusterDeployment) {
+				mockListComputeImage(client, []string{"testImage"}, cd.Status.InfraID)
```

Comment:
for the current list of test cases yes, but i was trying to handle someone coming in later and providing a non-standard clusterDeployment for a new test case.
but if it's premature optimization, i can pull it back to just use the global defaults

Reply:
The non-standard clusterdeployment does not change what the infra ID or region would be. In other words, there is no point in testing a cd with different values since the code just passes it along with any manipulation.